### ISSUE TITLE
Help:  How to define the  nn  parameters in torch.geometric.nn.nnConv?

### ISSUE THREAD

**Issue by Youarerare** [rusty1s/pytorch_geometric#1203]
## ❓ Questions & Help

What confuses me is the following in the NNConv document.

https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html?highlight=nnConv#torch_geometric.nn.conv.NNConv

> 'maps edge features edge_attr of shape [-1, num_edge_features] to shape [-1, in_channels * out_channels]

for example ,if I have a graph  have 3 node and 4 edge , node_embedding dim = 10,edge_embedding_dim = 5

> x =torch.randn(3,10,dtype = torch.float32)    #node embedding dim==10
> 
> edge_index = torch.tensor([[0, 1, 1, 2],[1, 0, 2, 1]], dtype=torch.long)     
>     
> edge_attr = torch.randn(4,5,dtype = torch.float32)  ###edge embedding dim ==5

how can I  define the NNConv in this graph? 
if I just want to encode the Node and edge information and return a New node representation. for example ,the new node embedding dim is 15.

> NNconv(in_channels, out_channels, nn, aggr='add', root_weight=True, bias=True, **kwargs)

What are these values of in_channels ,out_channels, and can the nn be a MLP or LSTM,how to define it?
Thank you

**Comment by rusty1s** [comment]
```python
mlp = Sequential(Linear(5, 5), ReLU(), Linear(5, 10 * 15) # maps to in_channels * out_channels
conv = NNConv(10, 15, mlp)
```
It converts each edge feature to a weight matrix, so you can use something like Linear layers or MLPs.

**Comment by Youarerare** [comment]
Thank you very much
